### PR TITLE
fix(server): Consume all upstream responses

### PR DESCRIPTION
**Pull request by jan-auer**
Relays do not reuse connections for store requests. The default response transformer for upstream requests does not consume the response payload, which is a potential cause for this.

cc @RaduW for https://github.com/getsentry/relay/pull/678

### CODE REVIEW

File: relay-server/src/middlewares.rs [hunk 1]
```diff
@@ -101,10 +101,12 @@ pub struct ReadRequestMiddleware;
...
+        let future = req
+            .payload()
+            .for_each(|_| Ok(()))
+            .map(|_| resp)
+            .map_err(Error::from);
 
-        Ok(Response::Future(Box::new(
-            consume_request.map(|_| resp).map_err(Error::from),
-        )))
+        Ok(Response::Future(Box::new(future)))
```
Comment (jan-auer) [inline relay-server/src/middlewares.rs:110]: This change is unrelated, just changed to have similar code style.

**Review by untitaker — APPROVED** [review]
how is this so buggy...

**Review by tonyo — APPROVED** [review]
Seems to work 🎉
